cheroot/test/webtest.py

```diff
@@ -478,42 +477,38 @@ def shb(response):
     return resp_status_line, h, response.read()
 
 
-def openURL(
-    url, headers=None, method='GET', body=None,
-    host='127.0.0.1', port=8000, http_conn=http_client.HTTPConnection,
-    protocol='HTTP/1.1', raise_subcls=None, ssl_context=None,
-):
+def openURL(*args, raise_subcls=(), **kwargs):
     """
-    Open the given HTTP resource and return status, headers, and body.
+    Open a URL, retrying when it fails.
 
-    `raise_subcls` must be a tuple with the exceptions classes
-    or a single exception class that are not going to be considered
-    a socket.error regardless that they were are subclass of a
-    socket.error and therefore not considered for a connection retry.
+    Specify `raise_subcls` (class or tuple of classes) to exclude
+    those socket.error subclasses from being suppressed and retried.
     """
-    headers = cleanHeaders(headers, method, body, host, port)
-
-    # Trying 10 times is simply in case of socket errors.
-    # Normal case--it should run once.
-    for trial in range(10):
-        try:
-            return _open_url_once(
-                body, headers, host, http_conn, method,
-                port, protocol, ssl_context, url,
-            )
-        except socket.error as e:
-            if raise_subcls is not None and isinstance(e, raise_subcls):
-                raise
-            else:
-                time.sleep(0.5)
-                if trial == 9:
-                    raise
+    opener = functools.partial(_open_url_once, *args, **kwargs)
+
+    def on_exception():
+        type_, exc = sys.exc_info()[:2]
+        if isinstance(exc, raise_subcls):
+            raise
+        time.sleep(0.5)
+
+    # Try up to 10 times
+    return jaraco.functools.retry_call(
+        opener,
+        retries=9,
+        cleanup=on_exception,
+        trap=socket.error,
+    )
 
 
 def _open_url_once(
-        body, headers, host, http_conn, method,
-        port, protocol, ssl_context, url,
+    url, headers=None, method='GET', body=None,
```

Comment:
pylint requires this to have more indentation. this is so it's not aligned with the function body.

Reply:
Linters seem to be passing for me and in CI. If the already nitpicking linters aren't complaining, I'm not going to worry about it.